### PR TITLE
Added docs for Offload Diagnostics and Controllable Fallback

### PR DESCRIPTION
**Pull request by 1e-to**
Docs for Offload diagnostics and controllable fallback.

### CODE REVIEW

File: docs/INDEX.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+Setting the debug environment variable `NUMBA_DPPY_OFFLOAD_DIAGNOSTICS `
+(e.g. `export NUMBA_DPPY_OFFLOAD_DIAGNOSTICS=1`) enables the parallel and offload diagnostics information.
+
+If set to an integer value between 1 and 4 (inclusive) diagnostic information about parallel transforms undertaken by Numba will be written to STDOUT. The higher the value set the more detailed the information produced.
+In the "Auto-offloading" section there is the information on which device (device name) this parfor or kernel was offloaded.
+
+## Controllable Fallback
+
+With the default behavior of numba-dppy, if a section of code cannot be offloaded on the GPU, then it is automatically executed on the CPU and printed a warning.
```
Comment (diptorupd) [inline docs/INDEX.md:15]: Please mention that this behavior only applies to `njit` functions and auto-offloading of numpy functions, array expressions, and prange loops.

**Comment by 1e-to** [comment]
Closes #67

**Comment by 1e-to** [comment]
@diptorupd , could you please check, that everything is ok now?

File: docs/INDEX.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
```
Comment (PokhodenkoSA) [inline docs/INDEX.md:15]:
```suggestion
With the default behavior of numba-dppy, if a section of code cannot be offloaded on the GPU, then it is automatically executed on the CPU and printed a warning. This behavior only applies to njit functions and auto-offloading of numpy functions, array expressions, and prange loops.
```

File: README.md [hunk 3]
```diff
@@ -67,3 +67,7 @@ Please follow instructions in the [DEBUGGING.md](DEBUGGING.md)
 ## Reporting issues
 
 Please use https://github.com/IntelPython/numba-dppy/issues to report issues and bugs.
+
+## Features
+
+Read this guide for additional features [INDEX.md](INDEX.md)
```
Comment (PokhodenkoSA) [inline README.md:73]:
```suggestion
Read this guide for additional features [INDEX.md](docs/INDEX.md)
```

File: docs/INDEX.md (hunk 1 above)
Comment (1e-to) [reply docs/INDEX.md:15]: Done at the end of the file.

**Review by PokhodenkoSA — APPROVED** [review]

**Pull request merged**